crates/uv-keyring/tests/threading.rs

```diff
@@ -0,0 +1,280 @@
+        assert!(
+            matches!(entry.get_password().await, Err(Error::NoEntry)),
+            "Able to read a deleted non-ascii password"
+        );
+    });
+
+    handle.await.expect("Task failed");
+}
+
+#[tokio::test]
+async fn test_simultaneous_create_then_move() {
+    init_logger();
+
+    let mut handles = vec![];
+    for i in 0..10 {
+        let name = format!("{}-{}", generate_random_string(), i);
+        let entry = Entry::new(&name, &name).expect("Can't create entry");
+
+        let handle = tokio::spawn(async move {
+            entry
+                .set_password(&name)
+                .await
+                .expect("Can't set ascii password");
+            let stored_password = entry
+                .get_password()
+                .await
+                .expect("Can't get ascii password");
+            assert_eq!(
+                stored_password, name,
+                "Retrieved and set ascii passwords don't match"
+            );
+            entry
+                .delete_credential()
+                .await
+                .expect("Can't delete ascii password");
+            assert!(
+                matches!(entry.get_password().await, Err(Error::NoEntry)),
+                "Able to read a deleted ascii password"
+            );
+        });
+        handles.push(handle);
+    }
+
+    for handle in handles {
+        handle.await.expect("Task failed");
+    }
+}
+
+#[tokio::test]
+#[cfg(not(target_os = "windows"))]
+async fn test_create_set_then_move() {
+    init_logger();
+
+    let name = generate_random_string();
+    let entry = Entry::new(&name, &name).expect("Can't create entry");
+    let password = "test ascii password";
+    entry
+        .set_password(password)
+        .await
+        .expect("Can't set ascii password");
+
+    let handle = tokio::spawn(async move {
+        let stored_password = entry
+            .get_password()
+            .await
+            .expect("Can't get ascii password");
+        assert_eq!(
+            stored_password, password,
+            "Retrieved and set ascii passwords don't match"
+        );
+        entry
+            .delete_credential()
+            .await
+            .expect("Can't delete ascii password");
+        assert!(
+            matches!(entry.get_password().await, Err(Error::NoEntry)),
+            "Able to read a deleted ascii password"
+        );
+    });
+
+    handle.await.expect("Task failed");
+}
+
+#[tokio::test]
+#[cfg(not(target_os = "windows"))]
+async fn test_simultaneous_create_set_then_move() {
+    init_logger();
+
+    let mut handles = vec![];
+    for i in 0..10 {
+        let name = format!("{}-{}", generate_random_string(), i);
+        let entry = Entry::new(&name, &name).expect("Can't create entry");
+        entry
+            .set_password(&name)
+            .await
+            .expect("Can't set ascii password");
+
+        let handle = tokio::spawn(async move {
```

Comment:
I'm not sure exactly what sort of mistakes this test is looking for, but I think wrapping blocking things in `spawn_blocking` will make it more faithful to the original without requiring any changes here?